### ISSUE TITLE
0.18.2 causes repated tracebacks in OpenStack

### ISSUE THREAD

**Issue by j2sol** [eventlet/eventlet#299]
I discovered this on a Kilo build, but looks like it's happening on master too.

https://bugs.launchpad.net/nova/+bug/1544801 is the bug I filed, even though it's against Nova it's happening on Neutron, Glance, maybe others.

```
2016-02-12 00:47:01.126 3936 DEBUG nova.api.openstack.wsgi [-] Calling method '<bound method Versions.index of <nova.api.openstack.compute.versions.Versions object at 0x7f05a0acfbd0>>' _process_stack /opt/bbc/openstack-11.0-bbc153/nova/local/lib/python2.7/site-packages/nova/api/openstack/wsgi.py:783
2016-02-12 00:47:01.129 3936 INFO nova.osapi_compute.wsgi.server [-] Traceback (most recent call last):
  File "/opt/bbc/openstack-11.0-bbc153/nova/local/lib/python2.7/site-packages/eventlet/wsgi.py", line 501, in handle_one_response
    write(b''.join(towrite))
  File "/opt/bbc/openstack-11.0-bbc153/nova/local/lib/python2.7/site-packages/eventlet/wsgi.py", line 442, in write
    _writelines(towrite)
  File "/opt/bbc/openstack-11.0-bbc153/nova/local/lib/python2.7/site-packages/eventlet/support/__init__.py", line 62, in safe_writelines
    writeall(fd, item)
  File "/opt/bbc/openstack-11.0-bbc153/nova/local/lib/python2.7/site-packages/eventlet/support/__init__.py", line 67, in writeall
    fd.write(buf)
  File "/usr/lib/python2.7/socket.py", line 324, in write
    self.flush()
  File "/usr/lib/python2.7/socket.py", line 303, in flush
    self._sock.sendall(view[write_offset:write_offset+buffer_size])
  File "/opt/bbc/openstack-11.0-bbc153/nova/local/lib/python2.7/site-packages/eventlet/greenio/base.py", line 383, in sendall
    tail = self.send(data, flags)
  File "/opt/bbc/openstack-11.0-bbc153/nova/local/lib/python2.7/site-packages/eventlet/greenio/base.py", line 377, in send
    return self._send_loop(self.fd.send, data, flags)
  File "/opt/bbc/openstack-11.0-bbc153/nova/local/lib/python2.7/site-packages/eventlet/greenio/base.py", line 364, in _send_loop
    return send_method(data, *args)
error: [Errno 104] Connection reset by peer
```

Further logs at 

http://logstash.openstack.org/#dashboard/file/logstash.json?query=message%3A%5C%22Traceback%20(most%20recent%20call%20last)%3A%5C%22%20AND%20message%3A%5C%22handle_one_response%5C%22%20AND%20tags%3A%5C%22screen%5C%22&from=7d

**Comment by mriedem** [comment]
Right, so this was due to b7380fdc70fde26777f4f141da3f01570e870cac which was meant to fix a problem with py3 and apparently has regressed something with python 2.7. We started seeing this on 2/4 across multiple projects (running eventlet / wsgi in their API service) and in both kilo and master (since they are uncapped and using latest eventlet, liberty is using 0.17.4 and doesn't have the problem).

Here is an example in the logs:

http://logs.openstack.org/22/277422/5/check/gate-tempest-dsvm-multinode-full/91079f7/logs/screen-n-api.txt.gz#_2016-02-11_13_29_09_861

**Comment by temoto** [comment]
Please check against eventlet master

`pip install https://github.com/eventlet/eventlet/archive/f66ab29cb57855ccb14ad84ef0534f9eb2bc573f.zip`

I'm eager to release it.

**Comment by j2sol** [comment]
Limited testing in Nova (Kilo) appears to work.

**Comment by temoto** [comment]
Fix is released in v0.18.3
